### ISSUE TITLE
Weird calculation of userSelectionRect on mouseMoving

### ISSUE THREAD

**Issue by Eatwhiterabbit** [wbkd/react-flow#1772]
Press shift-key, and then press the mouse, pulling a rectangle to the right. And still pressing the mouse, pull a rectangle to the left quickly.
Retry this procedure, will find out that the left-top point of the rectangle is not matched the first time triggered.

**Reproduction scenario**
https://reactflow.dev/examples/
need quickly switching the direction, or will not be reappeared.

**Maybe a better solution**
```
// originalCode
const nextUserSelectRect = {
  ...userSelectionRect,
  x: mousePos.x < startX ? mousePos.x : userSelectionRect.x,
  y: mousePos.y < startY ? mousePos.y : userSelectionRect.y,
  width: Math.abs(mousePos.x - startX),
  height: Math.abs(mousePos.y - startY),
};
```
Replaced as  ??
```
const nextUserSelectRect = {
  ...userSelectionRect,
  x: mousePos.x < startX ? mousePos.x : startX,
  y: mousePos.y < startY ? mousePos.y : startY,
  width: Math.abs(mousePos.x - startX),
  height: Math.abs(mousePos.y - startY),
};
```

thanks

**Comment by moklick** [comment]
very good point :D I will fix this. Thanks for the report!

**Comment by moklick** [comment]
released in v9.7.1
